### PR TITLE
LPD-18419 enable generative AI for product description

### PR DESCRIPTION
**Pull request by andrea-ale-sbarra**
https://liferay.atlassian.net/browse/LPD-18419

### CODE REVIEW

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/editor/configuration/CommerceCKEditorConfigContributor.java [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+			).buildString()
+		).put(
+			"aiCreatorPortletNamespace",
+			() -> _portal.getPortletNamespace(
+				"com_liferay_ai_creator_openai_web_internal_portlet_" +
+					"AICreatorOpenAIPortlet")
+		).put(
+			"extraPlugins", extraPlugins.concat(",aicreator")
+		).put(
+			"isAICreatorOpenAIAPIKey",
```
Comment (smottal) [inline modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/editor/configuration/CommerceCKEditorConfigContributor.java:85]: is it correct that starts with **is**? I think it's a string

File: modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/editor/configuration/CommerceCKEditorConfigContributor.java [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+			toJSONArray("['Link', Unlink]")
+		).put(
+			toJSONArray("['AICreator']")
+		);
+	}
+
+	private boolean _isAICreatorChatGPTGroupEnabled(
+		long companyId, long groupId) {
+
+		try {
```
Comment (smottal) [inline modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/editor/configuration/CommerceCKEditorConfigContributor.java:177]: Is a FF needed or is it already inside the isAICreatorChatGPTGroupEnabled method?

File: modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/editor/configuration/CommerceCKEditorConfigContributor.java [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+			toJSONArray("['Table', 'ImageSelector', 'VideoSelector']")
+		).put(
+			() -> {
+				if (AudioProcessorUtil.isEnabled()) {
+					return toJSONArray("['AudioSelector']");
+				}
+
+				return null;
+			}
+		).put(
```
[the remark this answers is not in the document]
Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/editor/configuration/CommerceCKEditorConfigContributor.java:143]: Screenshot 2024-03-04 at 16 51 21

File: modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/editor/configuration/CommerceCKEditorConfigContributor.java (hunk 1 above)
Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/editor/configuration/CommerceCKEditorConfigContributor.java:85]: I've copied pasted this from the webcontent editor, I think this is right

File: modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/editor/configuration/CommerceCKEditorConfigContributor.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/editor/configuration/CommerceCKEditorConfigContributor.java:143]: this I don't think should be changed

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/editor/configuration/CommerceCKEditorConfigContributor.java (hunk 2 above)
Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-product-definitions-web/src/main/java/com/liferay/commerce/product/definitions/web/internal/editor/configuration/CommerceCKEditorConfigContributor.java:178]: added it

File: portal-impl/src/portal.properties [hunk 4]
```diff
@@ -6895,6 +6900,11 @@
     #
     feature.flag.COMMERCE-12754.type=release
 
+    #
+    # Env: LIFERAY_FEATURE_PERIOD_FLAG_PERIOD__UPPERCASEL__UPPERCASEP__UPPERCASED__MINUS__NUMBER1__NUMBER0__NUMBER8__NUMBER6__NUMBER2__PERIOD_TYPE
+    #
+    feature.flag.LPD-10862.type=developer
+
```
Comment (smottal) [inline portal-impl/src/portal.properties:6907]: I think this is not needed because developer should be the default

File: test.properties
Comment (smottal) [inline test.properties:1]: Can you please move this file on a different commit because it's only SF

**Review by smottal — CHANGES_REQUESTED** [review]

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:forward:force

**Comment by andrea-ale-sbarra** [comment]
sent here https://github.com/brianchandotcom/liferay-portal/pull/148033
